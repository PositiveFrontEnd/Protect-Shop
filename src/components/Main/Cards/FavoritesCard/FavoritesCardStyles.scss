.favorite__card__wrapper {
  position: relative;
  display: flex;
  flex-direction: column;
  max-width: 320px;
  gap: 10px;

  @media (width>768px) {
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    max-width: 100%;
  }

  .favorite__card__image {
    position: relative;
    max-width: 320px;

    img {
      cursor: pointer;
    }

    @media (width>430px) {
      position: static;
    }

    @media (width>768px) {
      max-width: 160px;
    }

    @media (width>1024px) {
      max-width: 200px;
    }

    img {
      width: 100%;
    }

    .favorite__card__image-close {
      position: absolute;
      top: 1px;
      right: 10px;
      color: var(--inactive);
      font-size: 14px;

      @media (width>430px) {
        font-size: 18px;
      }
    }

    .favorite__card__image-close:hover {
      color: red;
      cursor: pointer;
    }
  }

  .favorite__card__name,
  .favorite__card__price {
    font-size: 16px;
    text-transform: uppercase;
    text-align: center;

    @media (width>768px) {
      font-size: 14px;
    }
  }

  .favorite__card__button {
    margin-right: 20px;
    font-size: 14px;
    display: flex;
    justify-content: center;
    width: 100%;

    @media (width>768px) {
      font-size: 12px;
      width: 20%;
    }
  }
}

.desktop__card__counter {
  display: flex;
  align-items: center;
  justify-content: space-around;

  .desktop__card__counter-min {
    width: 30px;
    height: 30px;
    background-color: #d9d9d9;
    border-radius: 10px;
  }
}
