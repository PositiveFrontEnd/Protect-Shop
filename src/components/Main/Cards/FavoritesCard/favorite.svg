<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<g id="Page-1">
<g id="add">
<path id="Combined-Shape" fill-rule="evenodd" clip-rule="evenodd" d="M13.724 0.390503L15.6096 2.27612L9.88569 7.99994L15.6096 13.7238L13.724 15.6095L8.00006 9.88556L2.27624 15.6095L0.390625 13.7238L6.1145 7.99994L0.390625 2.27612L2.27624 0.390503L8.00006 6.11438L13.724 0.390503Z" fill="#D3D8DA"/>
</g>
</g>
</svg>
