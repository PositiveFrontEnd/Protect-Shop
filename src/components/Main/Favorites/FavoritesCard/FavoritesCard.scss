.favorites__wrapper {
  .favorit__list__empty {
    color: var(--incorrect);
    text-align: center;
  }
}
