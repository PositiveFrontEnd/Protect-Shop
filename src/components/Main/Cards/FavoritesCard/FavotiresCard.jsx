import React from "react";
import "./FavoritesCardStyles.scss";
import Exit from './favorite.svg?react'
import { selectorToken } from "../../../../store/selectors";
import CounterForUser from './../PrimaryCard/ClickedCard/Desktop/CounterForUser';
import CounterForGuest from './../PrimaryCard/ClickedCard/Desktop/CounterForGuest';
import { useSelector } from "react-redux";
const FavorutesCard = ({ price, url, name, click, id, modalChangeAll, handleNavigate }) => {
  const token = useSelector(selectorToken)

  return (
    <div className="favorite__card__wrapper">
      <div className="favorite__card__image">
        <img src={url} alt="" onClick={handleNavigate} />
      </div>
      <span className="favorite__card__name">{name}</span>
      <span className="favorite__card__price">${price}</span>
      <div className="favorite__card__button">
        {token ? <CounterForUser _id={id} modalChangeAll={modalChangeAll} /> : <CounterForGuest _id={id} modalChangeAll={modalChangeAll} />}
      </div>
      <Exit onClick={click} className="favorite__card__image-close" />
    </div>
  );
};

export default FavorutesCard;
