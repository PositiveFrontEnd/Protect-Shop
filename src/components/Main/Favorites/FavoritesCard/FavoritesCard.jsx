import React from 'react'
import Button from '../../../Button/Button'
import './FavoritesCard.scss'

const FavoritesCard = ({ url, name, price }) => {
    return (
        <div className='favorite__item'>
            <div className='img__wrapper'>
                <img className='img' src={url} alt="" />
            </div>
            <p>{name}</p>
            <p>{price}</p>
            <Button black>Buy now</Button>

        </div>
    )
}
export default FavoritesCard