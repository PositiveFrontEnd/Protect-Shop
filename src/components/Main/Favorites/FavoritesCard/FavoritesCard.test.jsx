import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import FavoritesCard from "./FavoritesCard";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

describe("Тестування компонента  FavoritesCard", () => {
    test("Знімок", () => {
        const mockStore = configureStore([thunk]);
        const initialState = {

        };

        const store = mockStore(initialState);

        const data = {
            url: "https://chat.openai.com/c/d323d740-b8b7-417d-a832-84a6fe524b53",
            name: "some name",
            price: 1234
        };

        const favoritesCard = render(
            <Provider store={store}>
                <BrowserRouter>
                    <FavoritesCard data={data} />
                </BrowserRouter>
            </Provider>
        );

        expect(favoritesCard).toMatchSnapshot();
    });
});
