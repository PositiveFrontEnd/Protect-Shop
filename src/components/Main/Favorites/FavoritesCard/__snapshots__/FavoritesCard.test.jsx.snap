// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  FavoritesCard > Знімок 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="favorite__item"
      >
        <div
          class="img__wrapper"
        >
          <img
            alt=""
            class="img"
          />
        </div>
        <p />
        <p />
        <button
          class="button-black"
          type="button"
        >
          Buy now
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="favorite__item"
    >
      <div
        class="img__wrapper"
      >
        <img
          alt=""
          class="img"
        />
      </div>
      <p />
      <p />
      <button
        class="button-black"
        type="button"
      >
        Buy now
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
